wordpress-mobile/WordPress-iOS
# Update PostCategoriesViewController and WPCategoryTree to Swift

Updating `PostCategoriesViewController' to Swift as prep for implementing Categories in prepublishing bottom sheet.
 
## How to test

The goal in testing here is to verify that the app behaves as it did before this change.
So:

## Existing categories
1. Open a post
2. Go to Post Settings 
3. See the list of expected categories 

## Select / Diselect Category
1. Open or create a post 
2. Go to Post Settings
3. Tap on Categories
4. Tap on unselected category
5. See the checkmark accessory 
6. Tap Back 
7. See that the Category was added to the list 
* To test unselecting category, repeat 1-3 but tap a selected category and see that the category was removed from the list.  

## Add Category
1. Open or create a post 
2. Go to Post Settings
3. Tap on Categories
4. Tap +
5. Type a category name
6. Tap Save
7. See that the Category was added to the list and is selected 

## Parent Category
1. Open or create a post 
2. Go to Post Settings
3. Tap on Categories
4. Tap +
5. Type a category name
6. Tap Parent Category
7. Change parent 
8. See selected parent in row
6. Tap Save
7. See that the Category was added to the list under the parent and is selected 

# Remove category
1. While you are in Post Settings -> Categories
1. Go to your sites wp-admin on the web
2. Go to Posts
3. Got to Categories 
4. Delete a category 
5. In the app, pull to refresh the list of categories 
6. See the list updating as expected 

Please test anything else you can think of 🙏
